src/software/jetson_nano/thunderloop.cpp

```diff
@@ -12,19 +23,64 @@
 #include "software/logger/logger.h"
 #include "software/world/robot_state.h"
 
+/**
+ * https://rt.wiki.kernel.org/index.php/Squarewave-example
+ * using clock_nanosleep of librt
+ */
+extern int clock_nanosleep(clockid_t __clock_id, int __flags,
+                           __const struct timespec* __req, struct timespec* __rem);
+
+/*
+ * The struct timespec consists of nanoseconds and seconds. If the nanoseconds
+ * are getting bigger than 1000000000 (= 1 second) the variable containing
+ * seconds has to be incremented and the nanoseconds decremented by 1000000000.
+ *
+ * @param ts timespec to modify
+ */
+static inline void tsnorm(struct timespec& ts)
+{
+    while (ts.tv_nsec >= static_cast<int>(NANOSECONDS_PER_SECOND))
+    {
+        ts.tv_nsec -= static_cast<int>(NANOSECONDS_PER_SECOND);
+        ts.tv_sec++;
+    }
+}
+
+/**
+ * https://gist.github.com/diabloneo/9619917
+ *
+ * @fn timespec_diff(struct timespec *, struct timespec *, struct timespec *)
+ * @brief Compute the diff of two timespecs, that is a - b = result.
+ * @param a the minuend
+ * @param b the subtrahend
+ * @param result a - b
+ */
```

Comment:
need high level comment on this? Also why isn't this in the header?

Reply:
added and moved